### PR TITLE
Add an InitializeTarget function that can be specialized for a given target.

### PR DESCRIPTION
**Pull request by advaitjain**
This will allow the unit tests to be run on additional targets that need some addiitonal initialization (for example corstone_300 from #46830).

This particular change is broken out from the Corstone PR #46830 to be able to have smaller more reviewable PRs.

In the past, we have added state to the DebugLog() and GetCurrentTimeTicks() functions as a way to avoid having an InitializeTarget function. With this change, we are deciding to go with an explicit intitialization step instead.

This change has added calls to tflite::InitializeTarget to the tests, benchmarks, and examples and converted the Arduino and SparkfunEdge to make use of this explicit initialization.

The changes for the Arduino and SparkfunEdge have not been tested on actual hardware.

Progress towards #46829
Fixes http://b/150808076

### CODE REVIEW

File: tensorflow/lite/micro/system_setup.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_LITE_MICRO_SYSTEM_SETUP_H_
+#define TENSORFLOW_LITE_MICRO_SYSTEM_SETUP_H_
+
+namespace tflite {
+
+// This should called during initialization of TFLM binaries and tests. It can
+// be specialized if there us a need for custom target-specific intialization.
```
Comment (mansnils) [inline tensorflow/lite/micro/system_setup.h:21]: nitpick: typo

**Review by mansnils — COMMENTED** [review]
Looking good! Just a minor comment

File: tensorflow/lite/micro/system_setup.h (hunk 1 above)
Comment (advaitjain) [reply tensorflow/lite/micro/system_setup.h:21]: fixed.

**Comment by advaitjain** [comment]
@njeffrie: after our dicussion, I decided to go one step further and not add calls to InitializeTarget from DebugLog for the Arduino and SparkfunEdge.

Instead, I added a call to InitializeTarget from all of our examples. I like this approach more because it gives full control of the initialization to the application rather than having some mysterious init calls hidden away in DebugLog.

The down-side is that I might have broken some examples on Arduino/SparkfunEdge because the change is not tested on actual hardware.

**Review by njeffrie — APPROVED** [review]
We will have to update internal examples as well once this has landed.

**Pull request merged**